src/general/world_effects/patch_events/GlobalGlaciationEvent.cs

```diff
@@ -0,0 +1,218 @@
+using System.Collections.Generic;
+using Godot;
+using Newtonsoft.Json;
+using Xoshiro.PRNG64;
+
+[JSONDynamicTypeAllowed]
+public class GlobalGlaciationEvent : IWorldEffect
+{
+    private const string TemplateBiomeForIceChunks = "ice_shelf";
+    private const string Prefix = "globalGlaciation_";
+    private const string Background = "iceshelf";
+
+    private static readonly string[] IceChunksConfigurations =
+        ["iceShard", "iceChunkSmall", "iceChunkBig", "iceSnowflake"];
+
+    [JsonProperty]
+    private readonly XoShiRo256starstar random;
+
+    [JsonProperty]
+    private readonly bool hasEventAlreadyHappened;
+
+    [JsonProperty]
+    private readonly Dictionary<int, Dictionary<Compound, float>> previousEnvironmentalChanges = new();
+
+    [JsonProperty]
+    private readonly Dictionary<int, string> previousBackground = new();
+
+    [JsonProperty]
+    private readonly Dictionary<int, Color> previousLightColour = new();
+
+    /// <summary>
+    /// Tells how many generations the event will last. "-1" means that it hasn't started at all.
+    /// "0" means it has finished, and it won't happen again
+    /// </summary>
+    [JsonProperty]
+    private int generationsLeft = -1;
+
+    [JsonProperty]
+    private GameWorld targetWorld;
+
+    public GlobalGlaciationEvent(GameWorld targetWorld, long randomSeed)
+    {
+        this.targetWorld = targetWorld;
+        random = new XoShiRo256starstar(randomSeed);
+    }
+
+    [JsonConstructor]
+    public GlobalGlaciationEvent(GameWorld targetWorld, XoShiRo256starstar random)
+    {
+        this.targetWorld = targetWorld;
+        this.random = random;
+    }
+
+    public void OnRegisterToWorld() { }
+
+    public void OnTimePassed(double elapsed, double totalTimePassed)
+    {
+        if (generationsLeft > 0)
+            generationsLeft -= 1;
+
+        if (generationsLeft > 0)
+            MarkPatches(totalTimePassed);
+
+        if (hasEventAlreadyHappened)
+            return;
+
+        if (generationsLeft == -1)
+            TryToTriggerEvent(totalTimePassed);
+        else if (generationsLeft == 0)
+            FinishEvent();
+    }
+
+    private void MarkPatches(double totalTimePassed)
+    {
+        foreach (var patch in targetWorld.Map.Patches.Values)
+        {
+            if (Constants.SurfaceBiomes.Contains(patch.BiomeType))
+            {
+                patch.AddPatchEventRecord(WorldEffectVisuals.GlaciationEvent, totalTimePassed);
+            }
+        }
+    }
+
+    private void TryToTriggerEvent(double totalTimePassed)
+    {
+        if (!AreConditionsMet())
+            return;
+
+        generationsLeft = random.Next(2, 6);
```

Comment:
I think the duration range would be interesting to allow tweaking through `Constants`